jobs/build-podman.Jenkinsfile

```diff
@@ -0,0 +1,379 @@
+                    }
+                }]}
+            }
+
+            stage("Build Disk Images") {
+                  parallel archinfo.keySet().collectEntries{arch -> [arch, {
+                        if (arch != "x86_64") {
+		            pipeutils.withOptionalExistingCosaRemoteSession(
+                                arch: arch, session: archinfo[arch]['session']) {
+                                generate_disk_images(artifacts, shortcommit, arch, params.CONTAINER_REGISTRY_STAGING_REPO)
+		            }
+                        } else {
+                            generate_disk_images(artifacts, shortcommit, arch, params.CONTAINER_REGISTRY_STAGING_REPO)
+                        }
+                   }]}
+	    }
+
+           stage("Kola") {
+               parallel archinfo.keySet().collectEntries{arch -> [arch, {
+                   if (arch != "x86_64") {
+                       pipeutils.withOptionalExistingCosaRemoteSession(
+                           arch: arch, session: archinfo[arch]['session']) {
+                           shwrap("""cosa kola run  --basic-qemu-scenarios --qemu-image=./podman-${shortcommit}.qcow2""")
+                       }
+                   } else {
+                        shwrap("""cosa kola run  --basic-qemu-scenarios --qemu-image=./podman-${shortcommit}.qcow2""")
+                   }
+               }]}
+           }
+
+           stage("Compress Artifacts") {
+               parallel archinfo.keySet().collectEntries{arch -> [arch, {
+                    if (arch != "x86_64") {
+                        def session =  archinfo[arch]['session']
+                        pipeutils.withOptionalExistingCosaRemoteSession(
+                        arch: arch, session: session) {
+                            parallel artifacts.collectEntries{artifact -> [artifact["platform"] , {
+                                def file = "podman-$arch-${shortcommit}.${artifact["suffix"]}"
+                                shwrap("""podman --remote exec -i $session /usr/bin/zstd -14 $file""")
+                            }]}
+                        }
+                    } else {
+                       parallel artifacts.collectEntries{artifact -> [artifact["platform"] , {
+                           def file = "podman-$arch-${shortcommit}.${artifact["suffix"]}"
+                           shwrap("""/usr/bin/zstd -14 $file""")
+                       }]}
+                    }
+               }]}
+           }
+
+           stage("Push Manifest") {
+
+                //  Supermin is unable to access the secret mounted in another directory.
+                // Although this solution is not ideal, let's proceed with it until we
+                // can find a more effective approach.
+                // This doc includes a 'use' case that didn't function as expected.
+                // https://www.jenkins.io/doc/pipeline/steps/credentials-binding/
+                shwrap("""cp ${REGISTRY_SECRET} push-secret""")
```

Comment:
So actually, we do have a nice wrapper for this: [`utils.syncCredentialsIfInRemoteSession()`](https://github.com/coreos/coreos-ci-lib/blob/cf8e6a14b2c8b6d83940aa3f20ab6847345fd379/vars/utils.groovy#L65). You can see sample usage in other code in this repo.